adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -717,7 +689,34 @@ func createRoute(params *routeCreateParams) *routev3.Route {
 				Regex: routePath,
 			},
 		},
-		Headers: []*routev3.HeaderMatcher{&headerMatcherArray},
+	}
+
+	if !params.rewriteMethod {
+		// todo(amali) is this needed in enforcer
+		// OPTIONS is always added even if it is not listed under resources
+		// This is required to handle CORS preflight request fail scenario
+		methodRegex := strings.Join(resourceMethods, "|")
+		if !strings.Contains(methodRegex, "OPTIONS") {
```

Comment:
Wouldn't we need this even if this is a method rewrite?

Reply:
If the resource path has a method rewrite mediation, method validation will be handled in enforcer. Furthermore, added a code comment to explain it.